drivers/serial/Kconfig.gd32

```diff
@@ -7,7 +7,7 @@ DT_COMPAT_GIGADEVICE_GD32_USART := gd,gd32-usart
 config USART_GD32
 	bool "GD32 serial driver"
 	default $(dt_compat_enabled,$(DT_COMPAT_GIGADEVICE_GD32_USART))
-	depends on SOC_FAMILY_GD32
+	depends on SOC_FAMILY_GD32 && PINCTRL
```

Comment:
The SOC_FAMILY_GD32 config will always enable PINCTRL.

Reply:
done